### PR TITLE
Replace deploy TravisCI with GHA

### PR DESCRIPTION
**Pull request by apiraino**
I'd like to re-enable the deployment of the team compiler documentation website, which used to use TravisCI (see conversation [on Zulip](https://zulip-archive.rust-lang.org/stream/242791-t-infra/topic/Deploy.20of.20https.3A.2F.2Frust-lang.2Egithub.2Eio.2Fcompiler-team.html)), there are some updates that should be published.

I've pasted here the infra standard github action for static deploying on github.io, can we use that one?

perhaps @pietroalbini could you have a look here if I'm doing the right thing?

thanks!

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  push:
+    branches:
+      - master
+
+jobs:
+  deploy:
+    steps:
+      - uses: rust-lang/simpleinfra/github-actions/static-websites@master
+        with:
+          deploy_dir: /path/to/output
```
Comment (apiraino) [inline .github/workflows/deploy.yml:12]: @pietroalbini I'm not sure what to put here :thinking:

Comment (Xuanwo) [reply .github/workflows/deploy.yml:12]: Seems we need to use `$RUSTINFRA_DEPLOY_DIR`.

https://github.com/rust-lang/simpleinfra/blob/421f6a4429c8e6a60349ffd5f5192a89892ff08d/travis-configs/static-websites.yml#L28

Comment (apiraino) [reply .github/workflows/deploy.yml]: great suggestion, thanks! I've pushed the change

**Comment by apiraino** [comment]
@pietroalbini thanks for the tip, I have pushed some changes to the recipe, though I'm a bit blind here (we can't really test changes, can we?)

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository code
+      - uses: actions/checkout@v2
+      - name: Install Hugo
+      - run: |
+          wget -O /tmp/hugo.deb https://github.com/gohugoio/hugo/releases/download/v0.56.3/hugo_extended_0.56.3_Linux-64bit.deb
+          sudo dpkg -i /tmp/hugo.deb
+          hugo version
```
Comment (pierwill) [inline .github/workflows/deploy.yml:16]:
```suggestion
      - run: |
          HUGO_PATH="/tmp/hugo.deb"
          HUGO_VERSION="0.56.3"
          HUGO_SHA="4c8129b6f013d6b66c48446c681da8be20dfc504e5119c9f28a0b7115b9cb53e"
          wget -O ${HUGO_PATH} https://github.com/gohugoio/hugo/releases/download/v${HUGO_VERSION}/hugo_extended_${HUGO_VERSION}_Linux-64bit.deb
          echo "${HUGO_SHA}  ${HUGO_PATH}" | sha256sum --check && \
          sudo dpkg -i ${HUGO_PATH} && \
          hugo version
```

Comment (apiraino) [reply .github/workflows/deploy.yml:16]: cool, thank you @pierwill :+1:

**Pull request merged**